### ISSUE TITLE
issues 2021/2/11

### ISSUE THREAD

**Issue by ab-jxse** [chaupal/jxse#11]
After attributing the failing tests with @Ignore I found:
1. pom.xml creates jar file with source codes and I think it does not make sense. Here is the one that creates jar files with a) compiled classes and b) all classes with dependencies.
https://drive.google.com/file/d/1bdn804ayHbxlRrhx52zrWQNxK-UT9NJQ/view?usp=sharing
2. Classes in net.jxta.protocol: ResolverQueryMsg, ResolverResponseMsg refer net.jxse.JxseInstantiator. My understanding is that JXSE is built on the top of JXTA, so the latter should not depend on the former.
3. net.jxta.impl.peergroup.StdPeerGroup refers net.jxta.impl.util.Password which in turn refers Swing. I think Swing should not be used in JXTA at all. For example, if using JXTA on Android, Swing is not there. In general, all UI references should be eliminated in JXTA.

**Comment by ab-jxse** [comment]
Same as #2 with DynamicJxtaLoader and net.jxse.module.IJxtaModuleService. It seems that one of these classes need to move to a different place.
Same as #3 for net.jxta.impl.membership.pse.DialogAuthenticator. It looks that these authentications have to be addressed some other way.

**Comment by ab-jxse** [comment]
Kaas, can you create me a branch where I can upload my changes without sending them to you via Google Drive?

**Comment by condast** [comment]
I have given you write access to JXSE. 
You are not quite correct in your assumption that JXSE is built on top of JXTA. A bit of history is needed to understand this. The problem is that JXTA is a trademark of SUN/Oracle, and therefore we are strictly speaking not allowed to use the name. Around 2005 it was decided to adopt JXSE, in league with the JAVA-SE (Standard Edition) series. Former JXTA team lead Jerome Verstrynge also started to branch net.jxse in order to circumvent the poor open source license issues that was paralysing further development.  It was also a first step towards a more OSGI-based approach. The problem with the JXTA/JXSE sources is that it has become a monolith. I have tried to split up specification from implementation, but this opens up a can of worms, like you have also seen. JP2P aims to address this by creating a thin layer (based on Apache 2.0), which reflects the JXTA 2.0 specification document, behind which we can hide the current implementation. By enhancing modularity, we can start to phase out the current code behind the JP2P front end

**Comment by ab-jxse** [comment]
My short-term goal is to build a jar file that I can use with my applications, both on desktop and Android. So, which project can I use, this one or jp2p or jxse-osgi?
Regarding jxse/jxta fusion. I think it is confusing and shows bad design when different packages refer each other in loops. It defeats the purpose of packaging and should be addressed eventually.

**Comment by condast** [comment]
Currently I would go for jxse-2.8, because we are testing this one extesnively. jxse-osgi is almost the same (apart from the changes , we made lately) but needs testing on the spongycastle libs. These are implemented as OSGI fragments, and i'm not quite sure yet if they work without problems.

You are right about the design clutter, but because of the SUN/Oracle licensing no-one wants to make significant changes to the code, with the risk that Oracle takes the results as their own. That's why JP2P is underway

**Comment by condast** [comment]
I have pushed a new master branch:

1: removed dependencies to SWING. Instead, you can provide a password through the command line: -Dnet.jxta.stdpeergroup.password=myPassword

2: Fixed RefJxtaLoaderTest

3: Replaced Base64 implementation with that of apache

**Comment by ab-jxse** [comment]
Regarding #2. I do not think that storing credentials/passwords as the whole JVM-accessible property is a good idea. How about introducing an interface e.g. PasswordProvider?

**Comment by ab-jxse** [comment]
And there is a more serious issue with net.jxta.impl.membership.pse.DialogAuthenticator also used in PSEMembershipService

**Comment by condast** [comment]
Yep..but my first goal is to have JXTA run all the tests, and be able to build a version with all the dependencies that preferably runs on Android. with that the immediate aims have been satisfied, and the current code can be implemented on mobile. From then on I'll be able to phase out the old code and make improvements.

Anyway, the SWING dependencies were all at a fairly late stage of the authentication process, so the coee tries other alternatives first, e.g. as provided by NetworkConfigurator

**Comment by ab-jxse** [comment]
Tried to push my commit, got:
fatal: Authentication failed for 'https://github.com/chaupal/jxse.git/'
19:32:23.629: [jxse] git -c core.quotepath=false -c log.showSignature=false push --progress --porcelain origin refs/heads/ab-jxse:ab-jxse --set-upstream
remote: Permission to chaupal/jxse.git denied to ab-jxse.
fatal: unable to access 'https://github.com/chaupal/jxse.git/': The requested URL returned error: 403
19:36:21.637: [jxse] git -c core.quotepath=false -c log.showSignature=false push --progress --porcelain origin refs/heads/ab-jxse:ab-jxse --set-upstream
remote: Permission to chaupal/jxse.git denied to ab-jxse.
fatal: unable to access 'https://github.com/chaupal/jxse.git/': The requested URL returned error: 403
